File: backend/app/supabase_client.py
from functools import lru_cache

from supabase import Client, create_client

from .config import get_settings


@lru_cache
def get_supabase() -> Client:
    """Server-side Supabase client.

    Uses the service-role key, which bypasses Row Level Security — keep it
    server-only and never expose it to the browser.
    """
    settings = get_settings()
    if not settings.supabase_url or not settings.supabase_service_role_key:
        raise RuntimeError(
            "Missing Supabase config. Set SUPABASE_URL and "
            "SUPABASE_SERVICE_ROLE_KEY in backend/.env (copy from .env.example)."
        )
    return create_client(settings.supabase_url, settings.supabase_service_role_key)

File: frontend/src/lib/supabaseClient.ts
import { createClient } from '@supabase/supabase-js'

const supabaseUrl = import.meta.env.VITE_SUPABASE_URL
const supabaseAnonKey = import.meta.env.VITE_SUPABASE_ANON_KEY

if (!supabaseUrl || !supabaseAnonKey) {
  throw new Error(
    'Missing Supabase env vars. Set VITE_SUPABASE_URL and VITE_SUPABASE_ANON_KEY in frontend/.env (copy from .env.example).',
  )
}

// Browser-safe client: uses the public anon/publishable key only.
// Never expose the service-role key in frontend code.
export const supabase = createClient(supabaseUrl, supabaseAnonKey)
